subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/RoleBasedConfigurationContainerInternal.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.configurations;
+
+import org.gradle.api.Action;
+import org.gradle.internal.Actions;
+
+/**
+ * Extends {@link ConfigurationContainerInternal} with methods that can use {@link ConfigurationRole}s to
+ * define the allowed usage of a configuration at creation time.
+ */
+public interface RoleBasedConfigurationContainerInternal extends ConfigurationContainerInternal {
+    /**
+     * Creates a new configuration in the same manner as {@link #createWithRole(String, ConfigurationRole, boolean)}
+     * using the role of {@link ConfigurationRoles#INTENDED_CONSUMABLE}.
+     */
+    ConfigurationInternal consumable(String name, boolean lockRole);
```

Comment:
How come these don't have a default implementation here? I don't see why these need to be implemented in the `DefaultConfigurationContainer`.

Reply:
Good point.  Just me moving things around.  I'll change this.